File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Community/SelectCommunityDTO.cs
using platform_core_service.Common.Entities.BaseEntity;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Community
{
    public class SelectCommunityDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public string OwnerId { get; set; } = null!;
        public string Name { get; set; } = null!;
        public string? Description { get; set; }
        public string? CommunityCoverPhotoUrl { get; set; }
        public string Slug { get; set; } = null!;
        public bool IsPrivate { get; set; }
        public DateTimeOffset? DateCreated { get; set; }
        public DateTimeOffset? DateModified { get; set; }

        /// <summary>
        /// The current user's role in this community.
        /// Populated server-side in GetPageAsync to avoid N+1 queries.
        /// Values: "OWNER", "MODERATOR", "MEMBER", "PENDING", "GUEST"
        /// </summary>
        public string? CurrentUserRole { get; set; }
    }
}

File: frontend/src/types/community/select-community-dto.ts
export interface SelectCommunityDTO {
    id: string;
    ownerId: string;
    name: string;
    description?: string;
    communityCoverPhotoUrl?: string;
    slug: string;
    isPrivate: boolean;
    requireContentApproval: boolean;
    memberCount: number;
    dateCreated?: string;
    dateModified?: string;
    currentUserRole: string;
}
